### ISSUE TITLE
Microprofile non-loose application updates aren't working

### ISSUE THREAD

**Issue by rajivnathan** [eclipse/codewind#332]
**Codewind version:** 0.3.0
**OS:** Any

**Che version:** 0.3.0
**IDE extension version:** any
**IDE version:** any
**Kubernetes cluster:**

**Description:**
If the liberty-maven-plugin is configured with `<looseApplication>false</looseApplication>` then incremental updates won't work. This is because `mvn compile` is called for all update scenarios but non-loose applications require `mvn package`. The default https://github.com/microclimate-dev2ops/javaMicroProfileTemplate template is configured with `<looseApplication>true</looseApplication>` so it's not a problem that most users would hit.

**Steps to reproduce:**
1. Create a java microprofile project in Codewind.
2. Update the liberty-maven-plugin configuration in the microclimate profile to  `<looseApplication>false</looseApplication>`

**Workaround:** 

Making changes to the pom.xml of the application will trigger a `mvn package` to occur and the changes will be reflected correctly.

**Comment by johnmcollier** [comment]
Is this because we're doing a `mvn compile` on incremental updates, rather than a `mvn package`?

**Comment by rajivnathan** [comment]
Yes, exactly. Which is what I noted in the description ;) We'd need to detect when <looseApplication>false</looseApplication> is used and use `mvn package` instead.
